packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -103,6 +103,31 @@ abstract class VideoPlayerPlatform extends PlatformInterface {
   Future<void> setMixWithOthers(bool mixWithOthers) {
     throw UnimplementedError('setMixWithOthers() has not been implemented.');
   }
+
+  /// Returns true if picture in picture is supported on the device.
+  Future<bool> isPictureInPictureSupported() {
+    throw UnimplementedError(
+        'isPictureInPictureSupported() has not been implemented.');
+  }
+
+  /// Prepare picture in picture by passing the location of the video player view
+  Future<void> preparePictureInPicture({
```

Comment:
Because not every video should support PiP. We created a method prepare that should be called if you want the video to be PiP enabled. In order to get that you need to know the location of the video player. and the correct aspect ratio. (In dart we do that with the key and calculate the location of that widget)

After that you could start/stop the pip manually. Or if you set enableStartPictureInPictureAutomaticallyFromInline to true in the prepare method. you could also leave the app and the PiP would start automatically.

There is indeed more work need to get everything configured. But we don't think that every video should have the playerLayer added. if the video will never allow PiP. That is there reason why it is split up